File: HackerHank/DrawingBook/DrawingBookSolution.py
#
# Complete the 'pageCount' function below.
#
# The function is expected to return an INTEGER.
# The function accepts following parameters:
#  1. INTEGER n
#  2. INTEGER p
#

def pageCount(n, p):
    if(n % 2 == 0):
        return min(int(p/2), int((n - p + 1) / 2))
    return min(int(p/2), int((n - p) / 2))

File: HackerHank/DrawingBook/DrawingBookSolution.cpp
#include <algorithm>
using namespace std;
int pageCount(int n, int p)
{
    int turnsNeeded = p / 2;
    int ifEven = 0;
    if (n % 2 == 0)
        ifEven = (n - p + 1) / 2;
    else
        ifEven = (n - p) / 2;

    return min(turnsNeeded, ifEven);
}
